File: flutter_app/lib/src/models/todo.dart
class Todo {
  final int id;
  final String title;
  final bool status;

  Todo({required this.id, required this.title, required this.status});

  Todo fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json["id"] as int,
      title: json["title"] as String,
      status: json["status"] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "title": title,
      "status": status,
    };
  }
}

File: go_backend/types/todo.go
package types

import "time"

type Todo struct {
	ID        uint      `json:"id"`
	Title     string    `json:"title"`
	Status    bool      `json:"status"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

type TodoService interface {
	GetAllTodos() ([]Todo, error)
	GetTodoById(id uint) (Todo, error)
	CreateTodo(title string) (Todo, error)
	UpdateTodoById(todo Todo) (Todo, error)
	DeleteTodoById(id uint) error
}
